src/components/CatalogOverlay/CatalogPlotComponent/CatalogPlotComponent.tsx

```diff
@@ -311,50 +354,61 @@ export class CatalogPlotComponent extends React.Component<WidgetProps> {
     @computed get numBinsX(): number {
         const widgetStore = this.widgetStore;
         const profileStore = this.profileStore;
+        if (!widgetStore || !profileStore || !widgetStore.xColumnName) {
+            return 10; // default fallback
+        }
         const coords = profileStore.get1DPlotData(widgetStore.xColumnName);
-        const nBinx = Math.ceil(Math.sqrt(coords.wcsData?.length));
+        const nBinx = Math.ceil(Math.sqrt(coords.wcsData?.length ?? 100));
```

Comment:
`coords.wcsData?.length ? Math.ceil(Math.sqrt(coords.wcsData.length)): DEFAULT_NUM_BINS` would be more understandable